astroid/nodes/_base_nodes.py

```diff
@@ -10,15 +10,42 @@
 from __future__ import annotations
 
 import itertools
-from collections.abc import Iterator
-from functools import cached_property
-from typing import TYPE_CHECKING, ClassVar
-
-from astroid.exceptions import AttributeInferenceError
+from collections.abc import Generator, Iterator
+from functools import cached_property, lru_cache, partial
+from typing import TYPE_CHECKING, Any, Callable, ClassVar, Optional, Union
+
+from astroid import bases, decorators, nodes, util
+from astroid.const import PY310_PLUS
+from astroid.context import (
+    CallContext,
+    InferenceContext,
+    bind_context_to_node,
+    copy_context,
+)
+from astroid.exceptions import (
+    AttributeInferenceError,
+    InferenceError,
+    NameInferenceError,
+)
+from astroid.interpreter import dunder_lookup
 from astroid.nodes.node_ng import NodeNG
+from astroid.typing import InferenceErrorInfo, InferenceResult
 
 if TYPE_CHECKING:
-    from astroid import nodes
+    from astroid.nodes.node_classes import AssignedStmtsPossibleNode, LocalsDictNodeNG
+
+    GetFlowFactory = Callable[
+        [
+            InferenceResult,
+            Optional[InferenceResult],
+            Union[nodes.AugAssign, nodes.BinOp],
+            InferenceResult,
+            Optional[InferenceResult],
+            InferenceContext,
+            InferenceContext,
+        ],
+        "list[partial[Generator[InferenceResult, None, None]]]",
```

Comment:
Since this is under `TYPE_CHECKING` we should be able to do this without the `"`

Reply:
[c644b53](https://github.com/pylint-dev/astroid/pull/2171/commits/c644b53ea04af5e87cdf1ba536823dbe96b29775)